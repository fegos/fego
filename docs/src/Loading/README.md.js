webpackJsonp([9],{469:function(a,n){a.exports={content:["article",["h1","\u52a0\u8f7d Loading"],["h3","\u7ec4\u4ef6\u63cf\u8ff0"],["ul",["li",["p","\u7528\u4e8e\u663e\u793a\u52a0\u8f7d\u4e2d\u72b6\u6001\uff0c\u6b64\u65f6\u5185\u5bb9\u4e0d\u53ef\u64cd\u4f5c"]]],["h3","\u793a\u4f8b\u4ee3\u7801"],["pre",{lang:"html",highlighted:'<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>Loading</span> <span class="token attr-name">loading</span> <span class="token attr-name">type</span><span class="token attr-value"><span class="token punctuation">=</span><span class="token punctuation">"</span>bar<span class="token punctuation">"</span></span> <span class="token attr-name">size</span><span class="token attr-value"><span class="token punctuation">=</span><span class="token punctuation">"</span>large<span class="token punctuation">"</span></span> <span class="token punctuation">/></span></span>\n<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>Loading</span> <span class="token attr-name">loading</span> <span class="token attr-name">type</span><span class="token attr-value"><span class="token punctuation">=</span><span class="token punctuation">"</span>circle<span class="token punctuation">"</span></span> <span class="token punctuation">></span></span>\n  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>p</span><span class="token punctuation">></span></span>\u8fd9\u662f\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>p</span><span class="token punctuation">></span></span>\n  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>p</span><span class="token punctuation">></span></span>\u8fd9\u662f\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>p</span><span class="token punctuation">></span></span>\n  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>p</span><span class="token punctuation">></span></span>\u8fd9\u662f\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>p</span><span class="token punctuation">></span></span>\n  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>p</span><span class="token punctuation">></span></span>\u8fd9\u662f\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>p</span><span class="token punctuation">></span></span>\n<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>Loading</span><span class="token punctuation">></span></span>\n<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>Loading</span> <span class="token attr-name">loading</span> <span class="token attr-name">type</span><span class="token attr-value"><span class="token punctuation">=</span>{&lt;Icon</span> <span class="token attr-name">name</span><span class="token attr-value"><span class="token punctuation">=</span><span class="token punctuation">"</span>error<span class="token punctuation">"</span></span> <span class="token punctuation">/></span></span>} >\n  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>p</span><span class="token punctuation">></span></span>\u8fd9\u662f\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>p</span><span class="token punctuation">></span></span>\n<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>Loading</span><span class="token punctuation">></span></span>'},["code",'<Loading loading type="bar" size="large" />\n<Loading loading type="circle" >\n  <p>\u8fd9\u662f\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9</p>\n  <p>\u8fd9\u662f\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9</p>\n  <p>\u8fd9\u662f\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9</p>\n  <p>\u8fd9\u662f\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9</p>\n</Loading>\n<Loading loading type={<Icon name="error" />} >\n  <p>\u8fd9\u662f\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9\u5185\u5bb9</p>\n</Loading>']],["h2","API"],["h3","props\u5217\u8868"],["table",["thead",["tr",["th","\u5c5e\u6027"],["th","\u8bf4\u660e"],["th","\u7c7b\u578b"],["th","\u9ed8\u8ba4\u503c"]]],["tbody",["tr",["td","loading"],["td","\u662f\u5426\u663e\u793a\u52a0\u8f7d\u4e2d"],["td","boolean"],["td","true"]],["tr",["td","type"],["td","\u56fe\u6807\u7c7b\u578b\uff0c\u53ef\u9009 'bar', 'circle', \u4e5f\u53ef\u4ee5\u81ea\u5b9a\u4e49\u56fe\u6807\uff0c\u6b64\u65f6\u9700\u81ea\u884c\u5b9e\u73b0\u56fe\u6807\u7684\u52a8\u6548"],["td","react node"],["td","'bar'"]],["tr",["td","size"],["td","\u56fe\u6807\u5c3a\u5bf8\uff0c\u53ef\u9009 'small', 'default', 'large'"],["td","string"],["td","'default'"]],["tr",["td","tip"],["td","\u989d\u5916\u663e\u793a\u7684\u6587\u6848"],["td","string"],["td"]],["tr",["td","delay"],["td","\u65f6\u95f4\u5ef6\u8fdf"],["td","number"],["td","0"]]]]],meta:{title:"Loading",subTitle:"\u52a0\u8f7d",filename:"src/Loading/README.md"},toc:["ul",["li",["a",{className:"bisheng-toc-h1",href:"#\u52a0\u8f7d-Loading",title:"\u52a0\u8f7d Loading"},"\u52a0\u8f7d Loading"]],["li",["a",{className:"bisheng-toc-h3",href:"#\u7ec4\u4ef6\u63cf\u8ff0",title:"\u7ec4\u4ef6\u63cf\u8ff0"},"\u7ec4\u4ef6\u63cf\u8ff0"]],["li",["a",{className:"bisheng-toc-h3",href:"#\u793a\u4f8b\u4ee3\u7801",title:"\u793a\u4f8b\u4ee3\u7801"},"\u793a\u4f8b\u4ee3\u7801"]],["li",["a",{className:"bisheng-toc-h2",href:"#API",title:"API"},"API"]],["li",["a",{className:"bisheng-toc-h3",href:"#props\u5217\u8868",title:"props\u5217\u8868"},"props\u5217\u8868"]]]}}});