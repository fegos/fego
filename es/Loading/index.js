'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _Icon = require('../Icon');

var _Icon2 = _interopRequireDefault(_Icon);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Loading = function (_React$Component) {
  _inherits(Loading, _React$Component);

  function Loading(props) {
    _classCallCheck(this, Loading);

    var _this = _possibleConstructorReturn(this, _React$Component.call(this, props));

    _this.state = {
      loading: props.loading
    };
    _this.timer = null;
    return _this;
  }

  Loading.prototype.componentWillReceiveProps = function componentWillReceiveProps(nextProps) {
    var _this2 = this;

    var loading = nextProps.loading,
        delay = nextProps.delay;

    if (loading !== this.props.loading) {
      if (loading) {
        if (this.timer) {
          clearTimeout(this.timer);
        }
        this.timer = setTimeout(function () {
          // 启动loading，根据设置的延迟时间来启动
          _this2.setState({
            loading: true
          });
          clearTimeout(_this2.timer);
        }, delay);
      } else {
        // 取消loading，直接取消，不设置延迟
        this.setState({
          loading: false
        });
      }
    }
  };

  Loading.prototype.renderBody = function renderBody() {
    var _classNames;

    var _props = this.props,
        type = _props.type,
        size = _props.size,
        tip = _props.tip,
        prefixCls = _props.prefixCls;


    var tipEl = null;
    if (tip) {
      tipEl = _react2.default.createElement(
        'span',
        null,
        tip
      );
    }
    var iconEl = null;
    var cls = (0, _classnames2.default)(prefixCls + '-icon', (_classNames = {}, _classNames[prefixCls + '-icon-sm'] = size === 'small', _classNames[prefixCls + '-icon-lg'] = size === 'large', _classNames));
    if (type === 'bar') {
      iconEl = _react2.default.createElement(
        'div',
        { className: cls },
        _react2.default.createElement('span', { className: 'bar' }),
        _react2.default.createElement('span', { className: 'bar bar-2' }),
        _react2.default.createElement('span', { className: 'bar bar-3' })
      );
    } else if (type === 'circle') {
      iconEl = _react2.default.createElement(
        'div',
        { className: cls },
        _react2.default.createElement(_Icon2.default, { name: 'loading', className: 'circle' })
      );
    } else if (_react2.default.isValidElement(type)) {
      iconEl = _react2.default.createElement(
        'div',
        { className: cls },
        type
      );
    }
    return _react2.default.createElement(
      'div',
      { className: prefixCls + '-ctn' },
      iconEl,
      tipEl
    );
  };

  Loading.prototype.render = function render() {
    var _classNames2;

    var _props2 = this.props,
        children = _props2.children,
        prefixCls = _props2.prefixCls;
    var loading = this.state.loading;

    return _react2.default.createElement(
      'div',
      { className: '' + prefixCls },
      loading ? this.renderBody() : null,
      _react2.default.createElement(
        'div',
        { className: (0, _classnames2.default)((_classNames2 = {}, _classNames2[prefixCls + '-container'] = true, _classNames2['is-loading'] = loading, _classNames2['f-cb'] = true, _classNames2))
        },
        children
      )
    );
  };

  return Loading;
}(_react2.default.Component);

Loading.defaultProps = {
  // 前缀
  prefixCls: 'ns-loading',
  // 状态
  loading: true,
  // 时间延迟
  delay: 0,
  // 文案
  type: 'bar',
  // icon 大小
  tip: '',
  // icon 大小
  size: 'default'
};
Loading.propTypes = {
  // 前缀
  prefixCls: _propTypes2.default.oneOf(['ns-loading']),
  // 状态
  loading: _propTypes2.default.bool,
  // 时间延迟
  delay: _propTypes2.default.number,
  // 文案
  tip: _propTypes2.default.string,
  // 默认的icon类型:'bar', 'circle', 或自定义
  type: _propTypes2.default.node,
  // icon 大小
  size: _propTypes2.default.oneOf(['default', 'small', 'large'])
};
exports.default = Loading;