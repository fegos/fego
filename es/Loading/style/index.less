@import "../../style/mixin/index";
@loading-prefix: ~"@{prefix}-loading";


@keyframes loading-bar {
  0% {
    transform: scaleY(1);
  }
  20% {
    transform: scaleY(0.5);
  }
  50% {
    transform: scaleY(1);
  }
  80% {
    transform: scaleY(1.2);
  }
  100% {
    transform: scaleY(1);
  }
}

.@{loading-prefix} {
  position: relative;
  &-ctn {
    position: absolute;
    top: 35%;
    left: 0;
    right: 0;
    transform: translateY(-50%);
    text-align: center;
    color: @primary-5;
    z-index: 1;
  }
  &-icon {
    .bar{
      display: inline-block;
      width: 5px;
      height: 20px;
      border-radius: 3px;
      background: @primary-5;
      animation: loading-bar 1s linear infinite;
      margin: 0 3px;
    }
    .circle {
      font-size: 18px;
      color: @primary-5;
    }
    .bar-1 {
    }
    .bar-2 {
      animation-delay: .2s;
    }
    .bar-3 {
      animation-delay: .4s;
    }
  }
  &-icon-sm {
    .bar {
      width: 3px;
      height: 16px;
      border-radius: 1px;
    }
    .circle {
      font-size: 14px;
    }
  }
  &-icon-lg {
    .bar {
      width: 7px;
      height: 24px;
      border-radius: 3px;
    }
    .circle {
      font-size: 22px;
    }
  }
  &-container {
    transition: all .3s @ease-in-out;
    &.is-loading {
      // filter: blur(.5px); // 打开会造成小晃动，先关闭吧
      opacity: .7;
      &:before {
        content: "";
        position: absolute;
        left: 0;
        right: 0;
        top: 0;
        bottom: 0;
        background-color: @disabled-color-dark;
        opacity: .7;
        transition: all .3s;
        z-index: 10;
      }
    }
  }
}
