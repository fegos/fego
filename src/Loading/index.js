import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Icon from '../Icon';

export default class Loading extends React.Component {
  static defaultProps = {
    // 前缀
    prefixCls: 'ns-loading',
    // 时间延迟
    delay: 0,
    // 文案
    type: 'bar',
    // icon 大小
    tip: '',
    // icon 大小
    size: 'default',
  }
  static propTypes = {
    // 前缀
    prefixCls: PropTypes.oneOf(['ns-loading']),
    // 时间延迟
    delay: PropTypes.number,
    // 文案
    tip: PropTypes.string,
    // 默认的icon类型
    type: PropTypes.oneOf(['bar', 'circle']),
    // icon 大小
    size: PropTypes.oneOf(['default', 'small', 'large']),
  }
  constructor(props) {
    super(props);
    this.state = {
      loading: props.loading,
    };
    this.timer = null;
  }
  componentWillReceiveProps(nextProps) {
    const { loading, delay } = nextProps;
    if (loading !== this.props.loading) {
      if (loading) {
        if (this.timer) {
          clearTimeout(this.timer);
        }
        setTimeout(() => {
          // 启动loading，根据设置的延迟时间来启动
          this.setState({
            loading: true,
          });
          clearTimeout(this.timer);
        }, delay);
      } else {
        // 取消loading，直接取消，不设置延迟
        this.setState({
          loading: false,
        });
      }
    }
  }
  renderBody() {
    const {
      type, size, tip, prefixCls,
    } = this.props;

    let tipEl = null;
    if (tip) {
      tipEl = <span>{tip}</span>;
    }
    let iconEl = null;
    const cls = classNames(`${prefixCls}-icon`, {
      [`${prefixCls}-icon-sm`]: size === 'small',
      [`${prefixCls}-icon-lg`]: size === 'large',
    });
    if (type === 'bar') {
      iconEl = (
        <div className={cls} >
          <span className="bar" />
          <span className="bar bar-2" />
          <span className="bar bar-3" />
        </div>
      );
    } else if (type === 'circle') {
      iconEl = (
        <div className={cls} >
          <Icon name="loading" className="circle" />
        </div>
      );
    }
    return (
      <div className={`${prefixCls}-ctn`}>
        {iconEl}
        {tipEl}
      </div>
    );
  }
  render() {
    const { children, prefixCls } = this.props;
    const { loading } = this.state;
    return (
      <div className={`${prefixCls}`}>
        {loading ? this.renderBody() : null}
        <div className={classNames({
          [`${prefixCls}-container`]: true,
          'is-loading': loading,
          'f-cb': true,
        })}
        >
          {children}
        </div>
      </div>
    );
  }
}
