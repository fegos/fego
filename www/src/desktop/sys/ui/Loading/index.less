.page {
  padding: 20px 10px;
  :global {
    .section {
      margin-top: 20px;
      h3 {
        margin-bottom: 5px;
      }
      &.size {
        .ns-loading {
          width: 50px;
          height: 30px;
          display: inline-block;
        }
      }
    }
  }
}
