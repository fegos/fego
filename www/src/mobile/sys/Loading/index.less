.page {
  :global {
    .section {
      margin-bottom: 25px;
      &.size {
        .ns-loading {
          width: 50px;
          height: 30px;
          display: inline-block;
          white-space: normal;
          word-break: break-all;
          word-wrap: break-word;
        }
      }
    }
  }
}
