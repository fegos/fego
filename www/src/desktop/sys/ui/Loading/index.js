import React, { Component } from 'react';
import { Loading, Button } from 'fego';
import style from './index.less';

export default class Page extends Component {
  state = {
    loading: false,
  }
  render() {
    const ctn = (
      <div>
        <p>eqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweq</p>
        <p>eqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweq</p>
        <p>eqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweq</p>
        <p>eqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweqeqweq</p>
      </div>
    );
    return (
      <div className={style.page}>
        <div className="section">
          <p className="title">柱状加载中状态：</p>
          <Loading loading >
            {ctn}
          </Loading>
        </div>
        <div className="section">
          <p className="title">圆形加载中状态：</p>
          <Loading loading type="circle" >
            {ctn}
          </Loading>
        </div>
        <div className="section">
          <p className="title">带自定义文案的加载中状态：</p>
          <Loading loading type="bar" tip="正在拼命加载，别催，再催砍人了！！" >
            {ctn}
          </Loading>
        </div>
        <div className="section size">
          <p className="title">不同尺寸：</p>
          <Loading loading type="bar" size="small" />
          <Loading loading type="bar" />
          <Loading loading type="bar" size="large" />
          <Loading loading type="circle" size="small" />
          <Loading loading type="circle" />
          <Loading loading type="circle" size="large" />
        </div>
        <div className="section">
          <p className="title">带延迟：</p>
          <Button onClick={() => { this.setState({ loading: !this.state.loading }); }}>点击切换加载中状态</Button>
          <Loading delay={1000} loading={this.state.loading} >
            {ctn}
          </Loading>
        </div>
      </div>
    );
  }
}
