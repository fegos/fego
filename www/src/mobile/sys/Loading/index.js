import React, { Component } from 'react';
import { Loading, Button } from 'fego';
import style from './index.less';

export default class Page extends Component {
  state = {
    loading: false,
  }
  render() {
    const ctn = (
      <div>
        <p>静安寺看得见啊路上看见宽带连接达康书记看来大家卡洛斯建档立卡数据的拉克丝几大块垃圾堆拉看见角度看撒量较大克鲁赛德
        三代目三代目三，没懂你，阿森纳动漫的美少女们，按摩师，都那么，都那么三的面试，那
        </p>
      </div>
    );
    return (
      <div className={style.page}>
        <div className="section">
          <p className="title">柱状加载中状态：</p>
          <Loading loading >
            {ctn}
          </Loading>
        </div>
        <div className="section">
          <p className="title">圆形加载中状态：</p>
          <Loading loading type="circle" >
            {ctn}
          </Loading>
        </div>
        <div className="section">
          <p className="title">带自定义文案的加载中状态：</p>
          <Loading loading type="bar" tip="正在拼命加载，别催，再催砍人了！！" >
            {ctn}
          </Loading>
        </div>
        <div className="section size">
          <p className="title">不同尺寸：</p>
          <Loading loading type="bar" size="small" />
          <Loading loading type="bar" />
          <Loading loading type="bar" size="large" />
          <Loading loading type="circle" size="small" />
          <Loading loading type="circle" />
          <Loading loading type="circle" size="large" />
        </div>
        <div className="section">
          <p className="title">带延迟：</p>
          <Button onClick={() => { this.setState({ loading: !this.state.loading }); }}>点击切换加载中状态</Button>
          <Loading delay={1000} loading={this.state.loading} >
            {ctn}
          </Loading>
        </div>
      </div>
    );
  }
}
