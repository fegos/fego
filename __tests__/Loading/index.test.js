import React from 'react';
import Renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import Loading from 'Loading';

describe('Loading test', () => {
  const ctn = (
    <div>
      <p>文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容</p>
      <p>文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容</p>
      <p>文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容</p>
      <p>文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容</p>
    </div>
  );

  it('loading snapshot', () => {
    const loading = Renderer.create(<Loading loading>{ctn}</Loading>);
    expect(loading.toJSON()).toMatchSnapshot();
  });

  it('different loading type should render corrently', () => {
    const barLoading = shallow(<Loading loading type="bar">{ctn}</Loading>);
    expect(barLoading.find('.bar')).toHaveLength(3);
    expect(barLoading.find('.circle')).toHaveLength(0);

    const circleLoading = shallow(<Loading loading type="circle">{ctn}</Loading>);
    expect(circleLoading.find('.bar')).toHaveLength(0);
    expect(circleLoading.find('.circle')).toHaveLength(1);
  });

  it('element loading type should also render corrently', () => {
    const loading = shallow(<Loading loading type={<h3>for test</h3>}>{ctn}</Loading>);
    expect(loading.find('h3')).toHaveLength(1);
  });

  it('loading with tip should show the tip', () => {
    const loading = shallow(<Loading loading tip="test tip">{ctn}</Loading>);
    expect(loading.contains('test tip')).toBe(true);
  });

  it('component should response to props change, and render accordingly correct', () => {
    const loading = shallow(<Loading loading>{ctn}</Loading>);
    expect(loading.find('.ns-loading-ctn')).toHaveLength(1);
    loading.setProps({
      loading: false,
    });
    expect(loading.find('.ns-loading-ctn')).toHaveLength(0);
  });

  it('component with delay should response after the delay time', () => {
    jest.useFakeTimers();
    const loading = mount(<Loading loading={false} delay={1000}>{ctn}</Loading>);
    loading.setProps({
      loading: true,
    });
    expect(setTimeout).toHaveBeenCalledTimes(1);
    expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), 1000);
    /**
     * in order to trigger code in cocmponentWillRecieveProps
     * if (this.timer) {
          clearTimeout(this.timer);
        }
    */
    loading.setProps({
      loading: true,
    });
  });
});
