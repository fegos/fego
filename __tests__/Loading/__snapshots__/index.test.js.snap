// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading test loading snapshot 1`] = `
<div
  className="ns-loading"
>
  <div
    className="ns-loading-ctn"
  >
    <div
      className="ns-loading-icon"
    >
      <span
        className="bar"
      />
      <span
        className="bar bar-2"
      />
      <span
        className="bar bar-3"
      />
    </div>
  </div>
  <div
    className="ns-loading-container is-loading f-cb"
  >
    <div>
      <p>
        文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容
      </p>
      <p>
        文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容
      </p>
      <p>
        文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容
      </p>
      <p>
        文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容文本内容
      </p>
    </div>
  </div>
</div>
`;
